lib/mr_bump/slack.rb

```diff
@@ -16,31 +21,85 @@ def bump(version, changes)
       options = {}
       options[:icon_url] = @icon if @icon
       options[:attachments] = [attatchment(version, changes)]
-      notifier.ping ' ', options
+      notifier.ping 'There is a new version bump!', options
+    end
+
+    def notifier
+      @notifier ||= ::Slack::Notifier.new(@webhook, username: @username)
     end
 
     def user
-      [
-        ENV["USER"],
-        `whoami`
-      ].detect { |u| !u.nil? }
+      `git config user.name`.gsub(/\n/, '')
+    end
+
+    def git_url
+      @git_url ||= `git remote get-url origin`
+    end
+
+    def git_user
+      git_url[/[^@]+/].partition('//').last
+    end
+
+    def git_icon
+      'https://avatars.githubusercontent.com/' + git_user
+    end
+
+    def git_link
+      'https://github.com/' + git_user
     end
 
     def repo_name
       `git config --get remote.origin.url`[/[\w\.]+(?=\.git$)/]
     end
 
-    def notifier
-      @notifier ||= ::Slack::Notifier.new(@webhook, username: @username)
+    def jira_ids(changes)
+      changes.split('* ').map { |i| i.split(' - ',3)[1]}.compact
+    end
+
+    def jira_urls(changes)
+      jira_ids(changes).map do |ticket|
+        if ticket!= 'UNKNOWN'
+          '<' + @jira_url + '/browse/' + ticket + '|Jira link - ' + ticket + ">\n"
+        end
+      end.join
     end
 
+    def jira_field(changes)
+      if @jira_url != nil
+        {value: jira_urls(changes), short: true, unfurl_links: false}
+      end
+    end
+
+    def get_sha(id)
+      `git log --grep=#{id} --merges --format=format:%H`.split("\n").first
```

Comment:
Yeah, we can try and tweak the git command, can you do that so it only cares about things in the current branch? The only two cases I found with multiple were like this:
```
commit 4f41e7e61b57f978d69adeecadc9b8dc1959ac5e
Merge: 1f2c67b e9ce3ba
Date:   Thu Aug 18 12:54:06 2016 +0100

    Merge pull request #1192 from Intellection/hotfix/SP-1068

    Hotfix/SP-1068

commit e9ce3ba094a2ede9aad7638abc1db82fcb589d75
Merge: 2833bf2 1f2c67b
Date:   Thu Aug 18 11:55:35 2016 +0100

    Merge branch 'release/10.16.0' into hotfix/SP-1068

    # Conflicts:
    #   CHANGELOG.md
```
Obviously we only care about the merge into the `release/10.16.0` branch and not the reverse. Either way the one we actually care about is the most recent.